src/main/scala/com/cloudera/sparkts/models/ARIMAX.scala

```diff
@@ -476,7 +476,15 @@ class ARIMAXModel(
     var error = 0.0
 
     val xregMatrix = new BreezeDenseMatrix(rows = xreg.flatten.length / xreg.size, cols = xreg.size, data = xreg.flatten)
-    val xregPredictors = AutoregressionX.assemblePredictors(ts.toArray, MatrixUtil.matToRowArrs(xregMatrix), 0, xregMaxLag, includeOriginalXreg)
```

Comment:
Can you explain this change?  In what situations do we expect an exception from assemblePredictors?  And is there anyway to instead have a conditional for constructing the matrix one way over the other, rather than trying, failing, then trying the other way?